libraries/AP_InertialSensor/BatchSampler.cpp

```diff
@@ -87,8 +87,14 @@ void AP_InertialSensor::BatchSampler::periodic()
 
 void AP_InertialSensor::BatchSampler::update_doing_sensor_rate_logging()
 {
+    if (((batch_opt_t)(_batch_options_mask.get()) & BATCH_OPT_POST_FILTER)
+        && type == IMU_SENSOR_TYPE_GYRO) {
+        _doing_post_filter_logging = true;
+        return;
+    }
     if (!((batch_opt_t)(_batch_options_mask.get()) & BATCH_OPT_SENSOR_RATE)) {
         _doing_sensor_rate_logging = false;
+        _doing_post_filter_logging = false;
```

Comment:
why disable post filter logging here?

Reply:
We can't do post-filter, sensor rate batch sampling. So the intent is that these are exclusive and post-filter logging takes precedence. But there is a bug in this logic as we rotate between sensors which I have applied a fix for.